tools/tsp-client/src/typespec.ts

```diff
@@ -0,0 +1,154 @@
+import { NodeHost, compile, getSourceLocation } from "@typespec/compiler";
+import { parse, isImportStatement } from "@typespec/compiler";
+import { Logger } from "./log.js";
+import { readFile, readdir } from "fs/promises";
+import * as path from "node:path";
+import { parse as parseYaml } from "yaml";
+
+
+export async function getEmitterOptions(rootUrl: string, tempRoot: string, emitter: string, saveInputs: boolean, additionalOptions?: string): Promise<Record<string, any>> {
+  // TODO: Add a way to specify emitter options like Language-Settings.ps1, could be a languageSettings.ts file
+  let emitterOptions: Record<string, Record<string, unknown>> = {};
+  emitterOptions[emitter] = {};
+  if (additionalOptions) {
+    emitterOptions = resolveCliOptions(additionalOptions.split(","));
+  } else {
+    const configData = await readFile(path.join(tempRoot, "tspconfig.yaml"), "utf8");
+    const configYaml = parseYaml(configData);
+  
+    if (configYaml["options"] && configYaml["options"][emitter]){
+      emitterOptions[emitter] = configYaml["options"][emitter];
+      for (const key in emitterOptions[emitter]!) {
+        Object.keys(emitterOptions![emitter]!).forEach(
+          (k) => {
+            if (`{${k}}` === emitterOptions[emitter]![key]) {
```

Comment:
what is this trying to do? It almost looks like it's trying to replace the curly brace tokens in emitter options, but the logic doesn't really make sense to me

Reply:
There's a very common config in the tspconfig.yaml files where people are defining `package-name` by referencing their `{package-dir}` option. I updated this code and simplified it to only account for this case for now. 

Here is a tracking issue to investigate whether we need to do something beyond this simple replacement and work on a more generic solution: https://github.com/Azure/azure-sdk-tools/issues/7056